backend/ee/danswer/server/query_and_chat/chat_backend.py

```diff
@@ -120,17 +144,26 @@ def handle_simplified_chat_message(
     )
 
     response = ChatBasicResponse()
+    final_context_docs: list[LlmDoc] = []
 
     answer = ""
     for packet in packets:
         if isinstance(packet, DanswerAnswerPiece) and packet.answer_piece:
             answer += packet.answer_piece
         elif isinstance(packet, QADocsResponse):
-            response.simple_search_docs = translate_doc_response_to_simple_doc(packet)
+            response.simple_search_docs = _translate_doc_response_to_simple_doc(packet)
         elif isinstance(packet, StreamingError):
             response.error_msg = packet.error
         elif isinstance(packet, ChatMessageDetail):
             response.message_id = packet.message_id
+        elif isinstance(packet, FinalUsedContextDocsResponse):
```

Comment:
here we get the final context docs and get the int index of that doc in the list of docs we are already returning